servlet/src/main/java/com/linecorp/armeria/server/servlet/ServletHttpRequest.java

```diff
@@ -0,0 +1,1053 @@
+        this.dispatcher = dispatcher;
+    }
+
+    void setMultipartConfigElement(MultipartConfigElement multipartConfigElement) {
+        this.multipartConfigElement = multipartConfigElement;
+    }
+
+    void setServletSecurityElement(ServletSecurityElement servletSecurityElement) {
+        this.servletSecurityElement = servletSecurityElement;
+    }
+
+    void setAsyncSupportedFlag(Boolean asyncSupportedFlag) {
+        this.asyncSupportedFlag = asyncSupportedFlag;
+    }
+
+    /**
+     * Get servlet http exchange.
+     */
+    public ServletHttpExchange getServletHttpExchange() {
+        return servletHttpExchange;
+    }
+
+    /**
+     * Get netty request.
+     */
+    public AggregatedHttpRequest getHttpRequest() {
+        return httpRequest;
+    }
+
+    private Map<String, Object> getAttributeMap() {
+        return attributeMap;
+    }
+
+    /**
+     * Parse request scheme.
+     */
+    private void decodeScheme() {
+        scheme = httpRequest.scheme();
+    }
+
+    /**
+     * Parse area.
+     */
+    private void decodeLocale() {
+        final Locale[] locales;
+        final String headerValue = getHeader(HttpHeaderNames.ACCEPT_LANGUAGE.toString());
+        if (headerValue == null) {
+            locales = DEFAULT_LOCALS;
+        } else {
+            final String[] values = headerValue.split(",");
+            final int length = values.length;
+            locales = new Locale[length];
+            for (int i = 0; i < length; i++) {
+                final String value = values[i];
+                final String[] valueSp = value.split(";");
+                final Locale locale;
+                if (valueSp.length > 0) {
+                    locale = Locale.forLanguageTag(valueSp[0]);
+                } else {
+                    locale = Locale.forLanguageTag(value);
+                }
+                locales[i] = locale;
+            }
+        }
+        this.locales = locales;
+    }
+
+    /**
+     * Parsing coding.
+     */
+    private void decodeCharacterEncoding() {
+        String characterEncoding = ServletUtil.decodeCharacterEncoding(getContentType());
+        if (characterEncoding == null) {
+            characterEncoding = getServletContext().getRequestCharacterEncoding();
+        }
+       this.characterEncoding = characterEncoding;
+    }
+
+    /**
+     * parse parameter specification.
+     */
+    private void decodeBody(boolean bodyPartFlag) {
+        final Charset charset = Charset.forName(getCharacterEncoding());
+        final HttpDataFactory factory = getServletContext().getHttpDataFactory(charset);
+        int discardThreshold = 0;
+        if (multipartConfigElement != null) {
+            factory.setMaxLimit(multipartConfigElement.getMaxFileSize());
+            discardThreshold = multipartConfigElement.getFileSizeThreshold();
+        }
+
+        final HttpRequest request = new ServletMultipartRequest(httpRequest);
```

Comment:
I add new class ServletMultipartRequest here because Armeria HttpRequest is not support multipart. I will create another pull request to fix [#253 Add HTTP multipart content support](https://github.com/line/armeria/issues/253) and refactor this code later.